diff -up xf86-video-nv-2.1.20/src/nv_driver.c.starve xf86-video-nv-2.1.20/src/nv_driver.c
--- xf86-video-nv-2.1.20/src/nv_driver.c.starve	2012-07-18 03:43:02.000000000 -0400
+++ xf86-video-nv-2.1.20/src/nv_driver.c	2012-07-18 03:44:24.324393469 -0400
@@ -1279,6 +1279,12 @@ NVValidMode(SCRN_ARG_TYPE arg, DisplayMo
       if((pNv->fpWidth < mode->HDisplay) || (pNv->fpHeight < mode->VDisplay))
         return (MODE_PANEL);
 
+    /* avoid starving the chip on scanout */
+    if (pNv->Architecture == NV_ARCH_04 &&
+	(mode->HDisplay * mode->VDisplay * (pScrn->bitsPerPixel / 8) >
+	    pScrn->videoRam * 952)) /* 1024 * 0.93, yay magic numbers */
+	return MODE_BANDWIDTH;
+
     return (MODE_OK);
 }
 
